interfaces/apparmor/backend.go

```diff
@@ -855,6 +843,30 @@ func (b *Backend) addContent(securityTag string, snapInfo *snap.Info, cmdName st
 			return templateVariables(snapInfo, securityTag, cmdName)
 		case "###PROFILEATTACH###":
 			return fmt.Sprintf("profile \"%s\"", securityTag)
+		case "###FLAGS###":
+			// default flags
+			flags := []string{"attach_disconnected", "mediate_deleted"}
+			if spec.Unconfined() {
+				// need both parser and kernel support for unconfined
+				pfeatures, _ := parserFeatures()
+				kfeatures, _ := kernelFeatures()
+				if strutil.ListContains(pfeatures, "unconfined") &&
```

Comment:
do we need to ship a newer apparmor_parser version in snapd for this?

Reply:
No, the unconfined flag is in the version of apparmor that snapd is already vendoring. However the newer default_allow flag is not, and there are other new features as well. The plan is to release a new upstream version of apparmor in late Dec, early Feb, before FF and work on vendoring that.

Reply:
Re newer AppArmor in snapd - I created https://github.com/snapcore/snapd/pull/13354 for that purpose.